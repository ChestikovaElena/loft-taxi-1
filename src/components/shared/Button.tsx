import React from 'react';
import styled from 'styled-components';

const Button = ({ name }: { name: string }) => <StyledButton type="submit">{name}</StyledButton>;

export default Button;

const StyledButton = styled.button`
    align-self: flex-end;
    padding: 15px 50px;
    font-size: 21px;
    font-weight: 400;
    border: 0;
    border-radius: 4px;
    background-color: #ffc617;
`;
