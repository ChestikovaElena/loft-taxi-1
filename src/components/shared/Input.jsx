import React from 'react';
import styles from 'styled-components';

const StyledInput = styles.input`
    margin-bottom: 40px;
    width: 100%;
    padding: 8px 0;
    font-size: 16px;
    border: 0;
    border-bottom: 2px solid #eaeaea;
    transition: border-color 0.4s;
    
    &:hover {
        border-color: #1473e6;
    }
          
    &:focus {
        outline: none;
        border-color: #1473e6;
    }
`;

const Input = ({ type, id, placeholder }) => (
    <StyledInput type={type} id={id} placeholder={placeholder} />
);

export default Input;
