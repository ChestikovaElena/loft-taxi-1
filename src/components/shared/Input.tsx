import React from 'react';
import styled from 'styled-components';

const StyledInput = styled.input`
    margin-bottom: 40px;
    width: 100%;
    padding: 8px 0;
    font-size: 16px;
    border: 0;
    border-bottom: 2px solid #eaeaea;
    transition: border-color 0.4s;

    &:hover {
        border-color: #1473e6;
    }

    &:focus {
        outline: none;
        border-color: #1473e6;
    }
`;

const Input = ({ type, id, placeholder }: { type: string; id: string; placeholder: string }) => (
    <StyledInput type={type} id={id} placeholder={placeholder} />
);

export default Input;
