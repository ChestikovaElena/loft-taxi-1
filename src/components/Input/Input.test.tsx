import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';
import Button from './Button';

describe('Heading', () => {
    it('компонент рендерится без ошибок', () => {
        const input = shallow(<Input />);

        expect(input).toBeTruthy;
    });

    it('значения пропсов компонента отображаются корректно', () => {
        const input = shallow(
            <Input type="тип" id="идентификатор" placeholder="шаблонное описание" />,
        );

        expect(input.props()['type']).toBe('тип');
        expect(input.props()['id']).toBe('идентификатор');
        expect(input.props()['placeholder']).toBe('шаблонное описание');
    });
});
