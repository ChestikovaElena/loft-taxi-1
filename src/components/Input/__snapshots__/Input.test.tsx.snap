// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input компонент отображается корректно 1`] = `
<input
  className="sc-bdVaJa hzUHky"
  id="id"
  placeholder="описание"
  type="text"
/>
`;
